File: MercadoPagoSDKExamplesObjectiveC/MercadoPagoSDKExamplesObjectiveC/PaymentMethodPluginConfigViewController.m
//
//  PaymentMethodPluginConfigViewController.m
//  MercadoPagoSDKExamplesObjectiveC
//
//  Created by Juan sebastian Sanzone on 18/12/17.
//  Copyright © 2017 MercadoPago. All rights reserved.
//

#import "PaymentMethodPluginConfigViewController.h"

#import <MLUI/MLButton.h>
#import <MLUI/MLButtonStylesFactory.h>

@interface PaymentMethodPluginConfigViewController ()

@property (strong, nonatomic) PXPluginNavigationHandler * pluginNavigationHandler;
@property (weak, nonatomic) IBOutlet UILabel *messageLabel;
@property (weak, nonatomic) IBOutlet UITextField *walletTextField;

@end

@implementation PaymentMethodPluginConfigViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setupNextButton];
}

- (void)viewDidAppear:(BOOL)animated{
    [_walletTextField becomeFirstResponder];
    _messageLabel.text = nil;
}

#pragma mark - Setup methods
- (void)setupNextButton {
    MLButton *button = [[MLButton alloc] initWithConfig:[MLButtonStylesFactory configForButtonType:MLButtonTypePrimaryAction]];
    [button addTarget:self
               action:@selector(didTapOnNext)
     forControlEvents:UIControlEventTouchUpInside];
    button.buttonTitle = @"Continuar";
    button.frame = CGRectMake(0, 0, self.view.bounds.size.width, 55.0);
    _walletTextField.inputAccessoryView = button;
}


#pragma mark - Selectors/handlers
- (IBAction)didTapOnNext {
    
    if (self.pluginNavigationHandler != nil && [_walletTextField.text length] > 0) {
        _messageLabel.text = nil;
        [self.pluginNavigationHandler next];
    } else {
        _messageLabel.text = @"Debes completar este dato.";
    }
}

#pragma mark - Plugin implementation.

- (UIView * _Nullable)renderWithStore:(PXCheckoutStore * _Nonnull)store theme:(id<PXTheme> _Nonnull)theme {
    return self.view;
}

- (void)viewWillAppear {
    NSLog(@"PXPlugin viewWillAppear");
}

- (void)viewWillDisappear {
    NSLog(@"PXPlugin viewWillDisappear");
}

- (NSString * _Nullable)titleForNavigationBar {
    return @"Pagar con Bitcoin";
}

- (void)renderDidFinish {
    _messageLabel.text = nil;
    [_walletTextField becomeFirstResponder];
}

- (void)navigationHandlerForPluginWithNavigationHandler:(PXPluginNavigationHandler *)navigationHandler {
    self.pluginNavigationHandler = navigationHandler;
}

@end

File: PodTester/PodTester/Payment Method Plugins/PaymentMethodPluginConfigViewController.swift
//
//  PaymentMethodPluginConfigViewController.swift
//  PodTester
//
//  Created by Eden Torres on 12/12/17.
//  Copyright © 2017 AUGUSTO COLLERONE ALFONSO. All rights reserved.
//

import UIKit
import MercadoPagoSDK

class PaymentMethodPluginConfigViewController: UIViewController {

    fileprivate var navigationHandler: PXPluginNavigationHandler?

    @IBOutlet weak var passwordTextfield: UITextField!
    @IBOutlet weak var messageLabel: UILabel!

    override func viewDidLoad() {
        super.viewDidLoad()
        setupNextButton()
    }
}

// MARK: - Setup methods.
extension PaymentMethodPluginConfigViewController {

    fileprivate func setupNextButton() {
        let nextButton = UIButton(frame: CGRect(x: 0, y: 0, width: self.view.frame.width, height: 55.0))
        nextButton.backgroundColor = UIColor.mpDefaultColor()
        nextButton.tintColor = UIColor.white
        nextButton.setTitle("Continuar", for: .normal)
        nextButton.addTarget(self, action: #selector(PaymentMethodPluginConfigViewController.shouldNextAction), for: .touchUpInside)
        passwordTextfield.inputAccessoryView = nextButton
    }

    func shouldNextAction() {
        if let text = passwordTextfield.text, !text.isEmpty {
            navigationHandler?.next()
        } else {
            messageLabel.text = "Debes completar este campo."
        }
    }
}

// MARK: Plugin implementation.
extension PaymentMethodPluginConfigViewController: PXPluginComponent {

    func render() -> UIView {
        return self.view
    }

    func renderDidFinish() {
        messageLabel.text = nil
        passwordTextfield.text = nil
        passwordTextfield.becomeFirstResponder()
    }

    func titleForNavigationBar() -> String? {
        return "Pagar con Bitcoin"
    }

    func navigationHandlerForPlugin(navigationHandler: PXPluginNavigationHandler) {
        self.navigationHandler = navigationHandler
    }
}
